context/cms.js

```diff
@@ -12,7 +13,12 @@ let reducer = (state, action) => {
     case "set-records":
       return { ...state, records: action.payload };
     case "set-language":
-      return { ...state, language: action.payload };
+      const { records } = state;
+      return {
+        ...state,
+        language: action.payload,
+        records: processRecords(records, {language: action.payload} )
```

Comment:
since we pre-process records, we need to re-pre-process them on state change, nothing else will do it for us